ranger-core/src/main/java/io/appform/ranger/core/finderhub/ServiceFinderHub.java

```diff
@@ -169,4 +176,22 @@ private void updateRegistry() {
         }
         finders.set(updatedFinders);
     }
+
+    private void waitTillHubIsReady() {
+        serviceDataSource.services().forEach(service -> {
+            try {
+                val serviceRegistry = RetryerBuilder.<ServiceFinder<T, R>>newBuilder()
+                    .retryIfResult(r -> r == null)
+                    .build()
+                    .call(() -> getFinders().get().get(service));
+                RetryerBuilder.<Boolean>newBuilder()
+                    .retryIfResult(r -> r == null || !r)
+                    .build()
+                    .call(() -> serviceRegistry.getServiceRegistry().isRefreshed());
```

Comment:
Merge these two to one call. Also generally try to avoid multiple statements that can throw in a try block. I think sonar would have given you warnings on this already.

Reply:
Merged both using `Optional`